library/core/src/slice/iter.rs

```diff
@@ -1429,7 +1429,7 @@ impl<'a, T> DoubleEndedIterator for Windows<'a, T> {
     fn nth_back(&mut self, n: usize) -> Option<Self::Item> {
         let (end, overflow) = self.v.len().overflowing_sub(n);
         if end < self.size.get() || overflow {
-            self.v = &[];
+            self.v = &self.v[..0];
```

Comment:
I think it would make sense to also add the `// cheaper than &[]` comment to all these cases as well, it's not immediately obvious that it is